test/test_nn.py

```diff
@@ -10500,19 +10523,22 @@ def test_per_sample_weights_new_offsets(mode, dtype, trainable_scale, include_la
             expected.backward(grad)
             self.assertEqual(es.weight.grad, reference_weights.grad,
                              atol=dtype2prec_DONTUSE[dtype], rtol=0)
-            if trainable_scale:
+            if has_weight and trainable_scale:
                 self.assertEqual(per_sample_weights.grad, ref_per_sample_weights.grad,
                                  atol=dtype2prec_DONTUSE[dtype], rtol=0)
 
         if device == 'cuda':
             dtypes = (torch.float, torch.double, torch.half)
         else:
             dtypes = (torch.float, torch.double)
-        modes = ('sum',)
+        modes = ('sum', 'max', 'mean')
         trainable_scale = (True, False)
         include_last_offset = (True, False)
+        has_weight = (True, False)
```

Comment:
you probably wanted to include `has_weight` in the for loop iteration too

or you could do for simplicity:

```
modes = (('sum', False), ('sum', True), ('max', False), ('mean', False))

for dtype, (mode, has_weight), trainable, ...
```